// @yandex-cloud/nodejs-sdk/dist/generated/yandex/cloud/datatransfer/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ydb = exports.serializers = exports.postgres = exports.parsers = exports.mysql = exports.mongo = exports.kafka = exports.common = exports.clickhouse = exports.transfer_service = exports.transfer = exports.endpoint_service = exports.endpoint = void 0;
exports.endpoint = __importStar(require("./v1/endpoint"));
exports.endpoint_service = __importStar(require("./v1/endpoint_service"));
exports.transfer = __importStar(require("./v1/transfer"));
exports.transfer_service = __importStar(require("./v1/transfer_service"));
exports.clickhouse = __importStar(require("./v1/endpoint/clickhouse"));
exports.common = __importStar(require("./v1/endpoint/common"));
exports.kafka = __importStar(require("./v1/endpoint/kafka"));
exports.mongo = __importStar(require("./v1/endpoint/mongo"));
exports.mysql = __importStar(require("./v1/endpoint/mysql"));
exports.parsers = __importStar(require("./v1/endpoint/parsers"));
exports.postgres = __importStar(require("./v1/endpoint/postgres"));
exports.serializers = __importStar(require("./v1/endpoint/serializers"));
exports.ydb = __importStar(require("./v1/endpoint/ydb"));
