// @yandex-cloud/nodejs-sdk/dist/generated/yandex/cloud/organizationmanager/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.federation_service = exports.federation = exports.certificate_service = exports.certificate = exports.user_service = exports.user_account = exports.ssh_certificate_service = exports.organization_service = exports.organization = exports.group_service = exports.group_mapping_service = exports.group_mapping = exports.group = void 0;
exports.group = __importStar(require("./v1/group"));
exports.group_mapping = __importStar(require("./v1/group_mapping"));
exports.group_mapping_service = __importStar(require("./v1/group_mapping_service"));
exports.group_service = __importStar(require("./v1/group_service"));
exports.organization = __importStar(require("./v1/organization"));
exports.organization_service = __importStar(require("./v1/organization_service"));
exports.ssh_certificate_service = __importStar(require("./v1/ssh_certificate_service"));
exports.user_account = __importStar(require("./v1/user_account"));
exports.user_service = __importStar(require("./v1/user_service"));
exports.certificate = __importStar(require("./v1/saml/certificate"));
exports.certificate_service = __importStar(require("./v1/saml/certificate_service"));
exports.federation = __importStar(require("./v1/saml/federation"));
exports.federation_service = __importStar(require("./v1/saml/federation_service"));
