// @yandex-cloud/nodejs-sdk/dist/generated/yandex/cloud/dataproc/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.manager_service = exports.manager_job_service = exports.manager_job = exports.subcluster_service = exports.subcluster = exports.resource_preset_service = exports.resource_preset = exports.job_service = exports.job = exports.common = exports.cluster_service = exports.cluster = void 0;
exports.cluster = __importStar(require("./v1/cluster"));
exports.cluster_service = __importStar(require("./v1/cluster_service"));
exports.common = __importStar(require("./v1/common"));
exports.job = __importStar(require("./v1/job"));
exports.job_service = __importStar(require("./v1/job_service"));
exports.resource_preset = __importStar(require("./v1/resource_preset"));
exports.resource_preset_service = __importStar(require("./v1/resource_preset_service"));
exports.subcluster = __importStar(require("./v1/subcluster"));
exports.subcluster_service = __importStar(require("./v1/subcluster_service"));
exports.manager_job = __importStar(require("./manager/v1/job"));
exports.manager_job_service = __importStar(require("./manager/v1/job_service"));
exports.manager_service = __importStar(require("./manager/v1/manager_service"));
