// @yandex-cloud/nodejs-sdk/dist/generated/yandex/cloud/datasphere/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.user = exports.secret = exports.project_service = exports.project = exports.dataset = exports.community_service = exports.community = exports.project_data_service = exports.node_service = exports.node_execution_error_details = exports.folder_budget_service = exports.app_token_service = void 0;
exports.app_token_service = __importStar(require("./v1/app_token_service"));
exports.folder_budget_service = __importStar(require("./v1/folder_budget_service"));
exports.node_execution_error_details = __importStar(require("./v1/node_execution_error_details"));
exports.node_service = __importStar(require("./v1/node_service"));
exports.project_data_service = __importStar(require("./v1/project_data_service"));
exports.community = __importStar(require("./v2/community"));
exports.community_service = __importStar(require("./v2/community_service"));
exports.dataset = __importStar(require("./v2/dataset"));
exports.project = __importStar(require("./v2/project"));
exports.project_service = __importStar(require("./v2/project_service"));
exports.secret = __importStar(require("./v2/secret"));
exports.user = __importStar(require("./v2/user"));
