// @yandex-cloud/nodejs-sdk/dist/generated/yandex/cloud/serverless/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.apigateway_connection_service = exports.apigateway_connection = exports.triggers_trigger_service = exports.triggers_trigger = exports.triggers_predicate = exports.mdbproxy_proxy_service = exports.mdbproxy_proxy = exports.functions_function_service = exports.functions_function = exports.containers_container_service = exports.containers_container = exports.apigateway_service = exports.apigateway = void 0;
exports.apigateway = __importStar(require("./apigateway/v1/apigateway"));
exports.apigateway_service = __importStar(require("./apigateway/v1/apigateway_service"));
exports.containers_container = __importStar(require("./containers/v1/container"));
exports.containers_container_service = __importStar(require("./containers/v1/container_service"));
exports.functions_function = __importStar(require("./functions/v1/function"));
exports.functions_function_service = __importStar(require("./functions/v1/function_service"));
exports.mdbproxy_proxy = __importStar(require("./mdbproxy/v1/proxy"));
exports.mdbproxy_proxy_service = __importStar(require("./mdbproxy/v1/proxy_service"));
exports.triggers_predicate = __importStar(require("./triggers/v1/predicate"));
exports.triggers_trigger = __importStar(require("./triggers/v1/trigger"));
exports.triggers_trigger_service = __importStar(require("./triggers/v1/trigger_service"));
exports.apigateway_connection = __importStar(require("./apigateway/websocket/v1/connection"));
exports.apigateway_connection_service = __importStar(require("./apigateway/websocket/v1/connection_service"));
