// @yandex-cloud/nodejs-sdk/dist/generated/yandex/cloud/containerregistry/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.scanner_service = exports.scanner = exports.scan_policy_service = exports.scan_policy = exports.repository_service = exports.repository = exports.registry_service = exports.registry = exports.lifecycle_policy_service = exports.lifecycle_policy = exports.ip_permission = exports.image_service = exports.image = exports.blob = void 0;
exports.blob = __importStar(require("./v1/blob"));
exports.image = __importStar(require("./v1/image"));
exports.image_service = __importStar(require("./v1/image_service"));
exports.ip_permission = __importStar(require("./v1/ip_permission"));
exports.lifecycle_policy = __importStar(require("./v1/lifecycle_policy"));
exports.lifecycle_policy_service = __importStar(require("./v1/lifecycle_policy_service"));
exports.registry = __importStar(require("./v1/registry"));
exports.registry_service = __importStar(require("./v1/registry_service"));
exports.repository = __importStar(require("./v1/repository"));
exports.repository_service = __importStar(require("./v1/repository_service"));
exports.scan_policy = __importStar(require("./v1/scan_policy"));
exports.scan_policy_service = __importStar(require("./v1/scan_policy_service"));
exports.scanner = __importStar(require("./v1/scanner"));
exports.scanner_service = __importStar(require("./v1/scanner_service"));
