// @yandex-cloud/nodejs-sdk/dist/generated/yandex/cloud/k8s/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.version_service = exports.version = exports.node_group_service = exports.node_group = exports.node = exports.maintenance = exports.cluster_service = exports.cluster = void 0;
exports.cluster = __importStar(require("./v1/cluster"));
exports.cluster_service = __importStar(require("./v1/cluster_service"));
exports.maintenance = __importStar(require("./v1/maintenance"));
exports.node = __importStar(require("./v1/node"));
exports.node_group = __importStar(require("./v1/node_group"));
exports.node_group_service = __importStar(require("./v1/node_group_service"));
exports.version = __importStar(require("./v1/version"));
exports.version_service = __importStar(require("./v1/version_service"));
