// @yandex-cloud/nodejs-sdk/dist/generated/yandex/cloud/billing/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.sku_service = exports.sku = exports.service_service = exports.service = exports.customer_service = exports.customer = exports.budget_service = exports.budget = exports.billing_account_service = exports.billing_account = exports.billable_object = void 0;
exports.billable_object = __importStar(require("./v1/billable_object"));
exports.billing_account = __importStar(require("./v1/billing_account"));
exports.billing_account_service = __importStar(require("./v1/billing_account_service"));
exports.budget = __importStar(require("./v1/budget"));
exports.budget_service = __importStar(require("./v1/budget_service"));
exports.customer = __importStar(require("./v1/customer"));
exports.customer_service = __importStar(require("./v1/customer_service"));
exports.service = __importStar(require("./v1/service"));
exports.service_service = __importStar(require("./v1/service_service"));
exports.sku = __importStar(require("./v1/sku"));
exports.sku_service = __importStar(require("./v1/sku_service"));
