// ydb-sdk-lite/dist/query/yql-query.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.YqlQuery = void 0;
/**
 * Execute yql query via scripting service.
 */
const debug_1 = __importDefault(require("debug"));
const bundle_1 = require("../../proto/bundle");
const common_1 = require("./common");
const debug = (0, debug_1.default)('ydb-sdk-lite:yql');
const ExecuteYqlResult = bundle_1.Ydb.Scripting.ExecuteYqlResult;
class YqlQuery {
    constructor(grpc, tablePathPrefix) {
        this.grpc = grpc;
        this.tablePathPrefix = tablePathPrefix;
    }
    async execute(inputQuery, parameters = {}) {
        const request = {
            script: (0, common_1.addTablePathPrefix)(inputQuery, this.tablePathPrefix),
            parameters: (0, common_1.buildTypedParams)(inputQuery, parameters),
        };
        debug(`Executing yql query: ${request.script}`);
        const response = await this.grpc.scriptingService.executeYql(request);
        const payload = (0, common_1.getQueryPayload)(response, request.script || '', parameters);
        const result = ExecuteYqlResult.decode(payload);
        return (0, common_1.convertResultToJs)(result);
    }
}
exports.YqlQuery = YqlQuery;
